synapse/handlers/sso.py

```diff
@@ -458,6 +466,24 @@ async def complete_sso_login_request(
                     request.getClientIP(),
                 )
                 new_user = True
+            else:
+                if self._sso_update_profile_information:
```

Comment:
This can be `elif self._sso_update_profile_information:`.

Reply:
See https://github.com/matrix-org/synapse/pull/10108/commits/748fcf5429eb63430a8a2eb8f2caad8529f66f86